### ISSUE TITLE
conflicting dependencies

### ISSUE THREAD

**Issue by blshkv** [beefproject/beef#1590]
The latest git version requires both rack-protection  ~> 2.0.1 and sinatra, which requires rack-protection ~> 1.4

```
Bundler could not find compatible versions for gem "rack-protection":
  In Gemfile:
    rack-protection (~> 2.0.1)

    sinatra was resolved to 1.4.8, which depends on
      rack-protection (~> 1.4)
```

**Comment by bcoles** [comment]
Perhaps try using an older version of Sinatra. `rack-protection` version  works with sinatra version `1.2.8`.

```diff
diff --git a/Gemfile b/Gemfile
index 17080ff..910210d 100644
--- a/Gemfile
+++ b/Gemfile
@@ -8,7 +8,7 @@
 
 gem 'eventmachine'
 gem 'thin'
-gem 'sinatra'
+gem 'sinatra', '~> 1.2'
 gem 'rack', '~> 1.6.5'
 gem 'rack-protection', '~> 2.0.1'
 gem 'em-websocket' # WebSocket support
```

**Comment by blshkv** [comment]
ok,  that should solve but, it looks like you need to investigate further, and figure out how both or any of two library the latest version could be used.

**Comment by bcoles** [comment]
Thanks for reporting this issue. Fixed in 07c5473edf4b90af78fd5a7759e4ccecc1139c31.

**Comment by blshkv** [comment]
```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rack (~> 1.6.5)

    sinatra was resolved to 1.2.9, which depends on
      rack (< 1.5, ~> 1.1)

    thin was resolved to 1.7.2, which depends on
      rack (< 3, >= 1)

Could not find gem 'rack (< 1.5, ~> 1.1)', which is required by gem 'thin', in any of the sources.
```

**Comment by blshkv** [comment]
I do not like you requirement to downgrade a package. Please find a better solution. Currently, thin neee to be downgraded too. This will cease more problems with other tools.

Please re open the bug. I do not have the lock file

**Comment by blshkv** [comment]
Can you upgrade to sinatra 2.0* insead? It requires rack-protection-2 as well.

**Comment by bcoles** [comment]
I'm not surprised that you don't like my solution. It's not a very good solution.

Sinatra cannot be upgraded to 2.x as it depends on Rack 2.x. #1424

**Comment by blshkv** [comment]
That is correct. Please migrate to rack 2.0.x, it blocks everything.

**Comment by bcoles** [comment]
Fixed in 0a415b22520057eef565e22620a8ce13dd07e440.

**Comment by blshkv** [comment]
ok, we getting there. The next error:
```
Bundler could not find compatible versions for gem "json":
  In Gemfile:
    json

    dm-serializer was resolved to 1.2.2, which depends on
      json (~> 1.6)

    uglifier was resolved to 2.7.2, which depends on
      json (>= 1.8.0)
```

**Comment by bcoles** [comment]
Yeah that's not likely to be fixed any time soon. BeEF needs to be updated to work with Rack 2.x first.

As you've noticed the BeEF dependency situation is a mess and is likely to remain that way for the foreseeable future. BeEF project has no active developers.

**Comment by blshkv** [comment]
Please re-open this bug report for tracking purposes. Somebody need to sort it out.

The next problem is the therubyracer package, see:
https://github.com/cowboyd/therubyracer/issues/447

Basically, I'm unable to compile it because it requires unsupported libv8 with old v8.

**Comment by bcoles** [comment]
The Ruby Racer is no longer a dependency. Please update.

**Comment by blshkv** [comment]
can you check one more dep:
```
Bundler could not find compatible versions for gem "json":
  In Gemfile:
    json

    dm-serializer was resolved to 1.2.2, which depends on
      json (~> 1.6)
```

There is no issue with dev-ruby/dm-serializer-0.10.0 (it is locked in the Gemfile.lock currently)
